src/shared/Microsoft.Git.CredentialManager/Application.cs

```diff
@@ -47,62 +50,60 @@ public void RegisterProvider(IHostProvider provider, HostProviderPriority priori
             {
                 _configurationService.AddComponent(configurableProvider);
             }
+
+            // If the provider has custom commands to offer then create a root command for the provider
+            if (provider is ICommandProvider cmdProvider)
+            {
+                var providerCommand = new ProviderCommand(Context, provider);
+                cmdProvider.ConfigureCommand(providerCommand);
+                _providerCommands.Add(providerCommand);
+            }
+
         }
 
         protected override async Task<int> RunInternalAsync(string[] args)
         {
-            string appName = Path.GetFileNameWithoutExtension(_appPath);
-
-            // Construct all supported commands
-            var commands = new CommandBase[]
+            var rootCommand = new RootCommand();
+
+            var getCommand = new GetCommand(Context, _providerRegistry);
+            var storeCommand = new StoreCommand(Context, _providerRegistry);
+            var eraseCommand = new EraseCommand(Context, _providerRegistry);
+            var configureCommand = new ConfigureCommand(Context, _configurationService);
+            var unconfigureCommand = new UnconfigureCommand(Context, _configurationService);
+
+            // Add standard commands
+            rootCommand.AddCommand(getCommand);
+            rootCommand.AddCommand(storeCommand);
+            rootCommand.AddCommand(eraseCommand);
+            rootCommand.AddCommand(configureCommand);
+            rootCommand.AddCommand(unconfigureCommand);
```

Comment:
nit: could this be of the pattern

```cs
rootCommand.AddCommand(new GetCommand(Context, _providerRegistry);
```

instead of creating locals that have no other purpose?

Reply:
Sure!